src/Altinn.AccessManagement.Persistence/ResourceMetadataRepository.cs

```diff
@@ -70,11 +71,12 @@ public async Task<AccessManagementResource> InsertAccessManagementResource(Acces
 
         private static AccessManagementResource GetAccessManagementResource(NpgsqlDataReader reader)
         {
+            ResourceType resourceType;
             return new AccessManagementResource
             {
                 ResourceId = reader.GetFieldValue<int>("resourceid"),
                 ResourceRegistryId = reader.GetFieldValue<string>("resourceregistryid"),
-                ResourceType = reader.GetFieldValue<string>("resourcetype"),
+                ResourceType = Enum.TryParse(reader.GetFieldValue<string>("resourcetype"), out resourceType) ? resourceType : ResourceType.Default,
```

Comment:
A little bit unsure if we should use the default value, or not return/process the resource at all and instead log the event?

Reply:
Good point. Its not possible to write a value other than the values available in enum to AM.Resource. If write fails, this line will never be executed. However when a value is written to AM.Resource that is not part of the ResourceType enum, this line will proceed without any info. It would be a good idea to log but i don't think we would need to return null in this case. Also i expect that RR will already fail with a anonymous ResourceType.